core/src/Kokkos_Array.hpp

```diff
@@ -435,6 +435,46 @@ KOKKOS_FUNCTION constexpr T const&& get(Array<T, N> const&& a) noexcept {
 }  // namespace Kokkos
 //</editor-fold>
 
+//<editor-fold desc="Support for range-based for loop">
+namespace Kokkos {
+
+template <class T, std::size_t N>
+KOKKOS_FUNCTION constexpr T const* begin(
+    [[maybe_unused]] Array<T, N> const& a) noexcept {
+  if constexpr (N == 0)
+    return nullptr;
+  else
+    return &a[0];
```

Comment:
Would the following work ?
```suggestion
  return a.data();
```

Noting that from https://en.cppreference.com/w/cpp/container/array/data:
> If [size()](https://en.cppreference.com/w/cpp/container/array/size) is ​0​, data() may or may not return a null pointer.

So you might discard my comment.

Reply:
Actually I like your idea better